src/base/process_util.c

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2008 Search Solution Corporation. All rights reserved by Search Solution.
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+
+/*
+ * process_util.c - functions for process manipulation
+ */
+
+#include "process_util.h"
+
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+
+int
+create_child_process (const char *const argv[], int wait_flag, const char *stdin_file, char *stdout_file, char *stderr_file,
+                      int *exit_status)
+{
+#if defined(WINDOWS)
+  int new_pid;
+  STARTUPINFO start_info;
+  PROCESS_INFORMATION proc_info;
+  BOOL res;
+  int i, cmd_arg_len;
+  char cmd_arg[1024];
+  BOOL inherit_flag = FALSE;
+  HANDLE hStdIn = INVALID_HANDLE_VALUE;
+  HANDLE hStdOut = INVALID_HANDLE_VALUE;
+  HANDLE hStdErr = INVALID_HANDLE_VALUE;
+
+  if (exit_status != NULL)
+    *exit_status = 0;
+
+  for (i = 0, cmd_arg_len = 0; argv[i]; i++)
+    {
+      cmd_arg_len += sprintf (cmd_arg + cmd_arg_len, "\"%s\" ", argv[i]);
```

Comment:
depending on the solution you choose for creating a dynamic sized argument, we may need to use 'snprintf'